### ISSUE TITLE
bug: sway.systemdIntegration results in error on 22.05

### ISSUE THREAD

**Issue by adament** [nix-community/home-manager#2994]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

When I try to build sway with systemdIntegration = true (the default) I get an error about sway-session.target not being built
```
building Nix...
building the system configuration...
these 7 derivations will be built:
  /nix/store/wrz3pf0mbx920x8lybra60z47fbpqd07-sway-session.target.drv
  /nix/store/pw3ghyb8cxj0qhznp5jyi8lxc7p1x8dn-home-manager-files.drv
  /nix/store/dlm8z83rrxza5xj4j7yxxvj805fj4118-home-manager-generation.drv
  /nix/store/jndyji49c2szvr53f33nc1kysld53zpl-unit-home-manager-adament.service.drv
  /nix/store/jmdnly4n2g8jrc7izljhdqk79f2p69z9-system-units.drv
  /nix/store/5c8hazxc0va2x396nppa4wzykzkqlq58-etc.drv
  /nix/store/vp4h4s71ybhyrqsh57x6hzlzxz7wk248-nixos-system-adament-nixos-22.05.379.08950a6e29c.drv
building '/nix/store/wrz3pf0mbx920x8lybra60z47fbpqd07-sway-session.target.drv'...
error: builder for '/nix/store/wrz3pf0mbx920x8lybra60z47fbpqd07-sway-session.target.drv' failed to produce output path for output 'out' at '/nix/store/wrz3pf0mbx920x8lybra60z47fbpqd07-sway-session.target.drv.chroot/nix/store/p1dbzz4v1bkpajs4mm8qfn5lhxyd54pc-sway-session.target'
error: 1 dependencies of derivation '/nix/store/pw3ghyb8cxj0qhznp5jyi8lxc7p1x8dn-home-manager-files.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dlm8z83rrxza5xj4j7yxxvj805fj4118-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jndyji49c2szvr53f33nc1kysld53zpl-unit-home-manager-adament.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jmdnly4n2g8jrc7izljhdqk79f2p69z9-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5c8hazxc0va2x396nppa4wzykzkqlq58-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vp4h4s71ybhyrqsh57x6hzlzxz7wk248-nixos-system-adament-nixos-22.05.379.08950a6e29c.drv' failed to build
```
This error disappears if I add systemdIntegration = false to my configuration. If I attempt to build the target file manually (with systemdIntegration=false) instead by adding the following to my configuration file I get the same error
```
systemd.user.targets.sway-session = {
  Unit = {
    Description = "sway compositor session";
    Documentation = [ "man:systemd.special(7)" ];
    BindsTo = [ "graphical-session.target" ];
    Wants = [ "graphical-session-pre.target" ];
    After = [ "graphical-session-pre.target" ];
  };
};
```

### Maintainer CC

@alexarice @sumnerevans @sebtm 

### System information

```markdown
this path will be fetched (0.06 MiB download, 0.30 MiB unpacked):
  /nix/store/yjgdgysbfm79s3a5xjawcvz2qyfgpiw6-bash-interactive-5.1-p16-dev
copying path '/nix/store/yjgdgysbfm79s3a5xjawcvz2qyfgpiw6-bash-interactive-5.1-p16-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.43, NixOS, 22.05 (Quokka), 22.05.379.08950a6e29c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"home-manager-22.05.tar.gz, nixos-22.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by adament** [comment]
I just tried declaring the service with a different name and I got the same error. So maybe the issue is actually in the systemd module?

**Comment by berbiche** [comment]
Related to https://github.com/nix-community/home-manager/issues/2846.

Is this issue only reproducible under the 22.05 stable branch?

**Comment by adament** [comment]
I just tried upgrading to unstable and master and I get the same error there.

**Comment by adament** [comment]
I am not sure what I did to change things, my configuration looks the same as before. But the issue has disappeared now, which makes me suspect that it was my fault all along.

**Comment by TLATER** [comment]
This isn't a unique case, as that discourse post shows. Maybe keep it open until we at least know the actual root cause?

**Comment by asheshambasta** [comment]
Ran into the same; this issue should be reopened.

**Comment by TLATER** [comment]
For reference, in the discourse case, this was resolved by simply updating home-manager (from 21.11 to 22.05 in that case, but I imagine a specific commit in the 22.05 branch fixed this).
